src/Pelagos/Bundle/AppBundle/Command/ColdStorageFlagCommand.php

```diff
@@ -0,0 +1,179 @@
+                    $newDatasetSubmission->setDatasetSubmissionReviewStatus();
+                    // Set filesize of original file in new submission.
+                    $newDatasetSubmission->setDatasetFileColdStorageArchiveSize($size);
+                    // Set hash of original file in new submission.
+                    $newDatasetSubmission->setDatasetFileColdStorageArchiveSha256Hash($hash);
+                    // Set the creator of this entity to the user known as 'system'.
+                    $newDatasetSubmission->setCreator($systemPerson);
+
+                    // Set options for a new replacement datafile of the supplied Cold-Storage stubfile.
+                    $newDatasetSubmission->setDatasetFileTransferStatus(DatasetSubmission::TRANSFER_STATUS_NONE);
+                    $newDatasetSubmission->setDatasetFileName(null);
+                    $newDatasetSubmission->setDatasetFileSize(null);
+                    $newDatasetSubmission->setDatasetFileSha256Hash(null);
+                    $newDatasetSubmission->setDatasetFileTransferType(DatasetSubmission::TRANSFER_TYPE_SFTP);
+                    $newDatasetSubmission->setDatasetFileUri($stubFileName);
+                    $newDatasetSubmission->setDistributionFormatName('Cold Storage archive - ' .
+                        $datasetSubmission->getDistributionFormatName());
+
+                    // This is normally null, unless manually set. Handle that gracefully.
+                    $newDatasetSubmissionId = $newDatasetSubmission->getId();
+                    $metadata = $entityManager->getClassMetaData(get_class($newDatasetSubmission));
+                    $newDatasetSubmissionIdGenerator = $metadata->idGenerator;
+                    if ($newDatasetSubmissionId !== null) {
+                        // Temporarily change the ID generator to AssignedGenerator.
+                        $metadata->setIdGenerator(new AssignedGenerator());
+                    }
+
+                    $entityManager->persist($newDatasetSubmission);
+                    $entityManager->flush($newDatasetSubmission);
+
+                    if ($newDatasetSubmissionId !== null) {
+                        $metadata->setIdGenerator($newDatasetSubmissionIdGenerator);
+                    }
+
+                    $dataset->setDatasetSubmission($newDatasetSubmission);
+                    $entityManager->persist($dataset);
+                    $entityManager->flush($dataset);
+
+                    // This is an ugly hack, having to re-persist, but there seems not to be a cleaner way...
+                    // Reset original status.
+                    try {
+                        $newDatasetSubmissionReflection = new \ReflectionClass($newDatasetSubmission);
+                        $statusReflection = $newDatasetSubmissionReflection->getProperty('status');
+                        $statusReflection->setACcessible(true);
```

Comment:
A**c**cessible?

Reply:
Luckily, method names are case insensitive.  Will fix as it looks silly.